ui_framework/doc_site/src/components/guide_section/guide_section.jsx

```diff
@@ -0,0 +1,67 @@
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import Slugify from '../../services/string/slugify';
+
+export class GuideSection extends Component {
+  constructor(props) {
+    super(props);
+    this.onClickSource = this.onClickSource.bind(this);
+  }
+
+  getId() {
+    return Slugify.one(this.props.title);
+  }
+
+  onClickSource() {
+    this.context.openCodeViewer(this.props.source);
+  }
+
+  componentWillMount() {
+    this.context.registerSection(this.getId(), this.props.title);
```

Comment:
`registerSection` shouldn't be on `context`, rather rely on `connect` to inject these into the component instead. I would create a `GuideSection` container that does it.

(`context` is an escape hatch, so prefer not to use it. In this case you'd still _actually_ depend on `context`, but through react-redux instead. I think that's okey, as it's a "stricter api")